compiler/optimizer/GlobalRegisterAllocator.cpp

```diff
@@ -800,29 +800,35 @@ TR_GlobalRegisterAllocator::restoreOriginalSymbol(TR::Node *node, vcount_t visit
             {
             TR_RegisterCandidate *origRc = _registerCandidates[origSymRef->getReferenceNumber()];
 
-            if (setValueModified)
-               _valueModifiedSymRefs->set(origRc->getSymbolReference()->getReferenceNumber());
+            if (origRc)
```

Comment:
Something doesn't seem right here. You've got a conditional structure:
 if (a || b) { if (c) { if (!a && d) ... 

Once you're under (a || b) isn't !b the same as "a" ?

But that makes me suspicious of the "if (!origRc->getValueModified() && origRc->getRestoreSymbolReference()" which could be rewritten as "if (origRc->extendedLiveRange() && origRc->getRestoreSymbolReference())". That feels like a strange condition here, but I didn't look at the whole context.

@andrewcraik or @vijaysun-omr, would you please take a closer look at this code to make sure there isn't a latent bug in the conditions that might be tweaked by this refactoring?